assets/app/views/EditorView.js

```diff
@@ -41,17 +44,46 @@ var EditorView = Backbone.View.extend({
 
     this.$('.js-st-instance').text(doc.toSirTrevorJsonString());
     this.editor.reinitialize();
-
+    $('form#metadata').hide();
     return this;
   },
+  toggleAreas: function () {
+    $('form#metadata').toggle();
+    $('form#content').toggle();
+  },
   saveDocument: function (e) {
+    var formFrontMatter = {};
     SirTrevor.onBeforeSubmit();
+    $('form#metadata .row').each(function(index, row) {
+      var key   = $(row).find('.front-matter-key').val(),
+          value = $(row).find('.front-matter-value').val();
+
+      if (key) {
+        formFrontMatter[key] = value;
+      }
+    });
+
+    this.doc.updateFrontMatter(formFrontMatter);
     this.doc.updateContentFromSirTrevorJson(this.editor.store.retrieve());
+
     this.trigger('edit:save', {
         md: this.doc.toMarkdown(),
         msg: this.$('#save-content-message').val()
     });
     return this;
+  },
+  deleteMetaDataRow: function (e) {
+    e.preventDefault();
+    $(e.target).parents('.meta-data-rows .row').remove();
+
+    return this;
+  },
+  addMetaDataRow: function (e) {
+    e.preventDefault();
+    var rowTemplate = _.template('<div class="row"><div class="col s5"><input type="text" class="front-matter-key" placeholder="key"></div><div class="col s5"><input type="text" class="front-matter-value" placeholder="value"></div><div class="col s2"><button class="front-matter-delete">Delete</button></div></div>')();
```

Comment:
Yes, I think I'll do that in the merge conflicts with #127, but it was hard to imagine how to name it in a sane way